spec/services/indexing_service_spec.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe IndexingService, type: :service do
+  let(:resource) { build(:work_version) }
+
+  describe '.call' do
+    subject(:service) { described_class.call(resource: resource, schema: {}) }
+
+    context 'with a supported resource' do
+      let(:resource) { instance_spy('IndexedResource', uuid: SecureRandom.uuid, class: 'IndexedResource') }
+
+      it 'indexes a minimal amount of metadata in Solr' do
+        service
+        document = SolrDocument.find(resource.uuid)
+        expect(document.id).to eq(resource.uuid)
+        expect(document['model_ssi']).to eq('IndexedResource')
+      end
+    end
+
+    context 'with an improperly identified resource' do
+      let(:resource) { Struct.new('UnsupportedResource') }
```

Comment:
Just FYI, `Struct.new` returns back a constant, not an instance.

Ideas to make this look more like a "resource" object...

```
Struct.new('UnsupportedResource').new
BasicObject.new
mock('UnsupportedResource')
```